packages/lit-dev-content/site/tutorials/content/reactivity/04.md

```diff
@@ -0,0 +1,9 @@
+One way your component can react to property changes is to compute values based on the new and existing property values.
+
+If you need to compute a value to render, you can compute it in the `render()` method itself. However a more general solution is to compute values in the `willUpdate()` callback. This has a couple of advantages:
```

Comment:
Let's lead with the right way: do this in `willUpdate`. Computing state (properties are are saved on the element) in render is not something we should generally recommend since we want to encourage `render` to just be for rendering and not side-effectful.

Computing ephemeral values needed just for rendering in `render` is a-ok, but stateful computed values should be done in willUpdate.

You typically want `changedProperties` to know if you want to compute and be able to see old values. You get this info in willUpdate, but not in render.

Reply:
Done.